### PR TITLE
Add preemptible machines boilerplate

### PR DESCRIPTION
**Pull request by jaspervandenbossche-ml6**
PR contains the boilerplate for automatically writing checkpoints to GCS while training ML models on preemptible machines. @timdesmetML6

### CODE REVIEW

**Comment by timdesmetML6** [comment]
Hi @jaxpr, totally lost track of this. @Ruwann can you add me as a reviewer please?

File: engineering/2022_04_25_preemptible_machines/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Training Machine Learning Models on Preemptible VMs
+
+Training a machine learning models on Google Cloud is not only computationally expensive but also expensive for your wallet 💰. But don't let this stop you from building ML applications. In this repository we will show you this on simple trick that allows you to save a lot of money! ...We promise it isn't a scam. Google offers its excess Compute Engine capacity as preemptible VM instances at much lower rates than standard VM instances.  There is a caviat though, Google can claim these machines back at any time, even in the middle of your computations 😵. To avoid losing all your  progress while training your model, you can save checkpoints to a Google Cloud Storage bucket. We have a Tensorflow and Pytorch example you can use as boilerplate. If you were to lose your preemptible VM, you can then restart training from your latest checkpoint.
```
Comment (timdesmetML6) [inline engineering/2022_04_25_preemptible_machines/README.md:3]: Small typo: no space between `...` and `We`

**Review by timdesmetML6 — APPROVED** [review]
Nice work!

**Comment by jaxpr** [comment]
Typo has been fixed!

**Review by Ruwann — COMMENTED** [review]
Thanks, @jaxpr !

Couple of comments
* Can you add this one to the list in the main engineering README as well?
* It's not entirely clear what the take-away is: these are some boilerplate functions I can use myself, and not a ready-to-go boilerplate that also takes care of the GCP parts?
* Could you clarify this in the README? Perhaps as a separate paragraph ?

File: engineering/2022_04_25_preemptible_machines/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Training Machine Learning Models on Preemptible VMs
+
+Training a machine learning models on Google Cloud is not only computationally expensive but also expensive for your wallet 💰. But don't let this stop you from building ML applications. In this repository we will show you this on simple trick that allows you to save a lot of money! ... We promise it isn't a scam. Google offers its excess Compute Engine capacity as preemptible VM instances at much lower rates than standard VM instances.  There is a caviat though, Google can claim these machines back at any time, even in the middle of your computations 😵. To avoid losing all your  progress while training your model, you can save checkpoints to a Google Cloud Storage bucket. We have a Tensorflow and Pytorch example you can use as boilerplate. If you were to lose your preemptible VM, you can then restart training from your latest checkpoint.
```
Comment (Ruwann) [inline engineering/2022_04_25_preemptible_machines/README.md:3]: typo:

`caviat` --> `caveat`

**Comment by jaxpr** [comment]
@Ruwann sorry for the late update, didn't receive a notification that you reviewed the PR, but I implemented your requests.

**Review by Ruwann — APPROVED** [review]

**Comment by Fematich** [comment]
I see this one is not merged yet, while approved, merging now!

**Pull request merged**